docker/resource_docker_plugin_funcs.go

```diff
@@ -0,0 +1,165 @@
+package docker
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"log"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/client"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func getDockerPluginArgs(src interface{}) []string {
+	if src == nil {
+		return nil
+	}
+	b := src.(*schema.Set)
+	args := make([]string, b.Len())
+	for i, a := range b.List() {
+		args[i] = a.(string)
+	}
+	return args
+}
+
+func resourceDockerPluginCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ProviderConfig).DockerClient
+	ctx := context.Background()
+	pluginRef := d.Get("plugin_reference").(string)
+	alias := d.Get("alias").(string)
+	log.Printf("[DEBUG] Install a Docker plugin " + pluginRef)
+	body, err := client.PluginInstall(ctx, alias, types.PluginInstallOptions{
+		RemoteRef:            pluginRef,
+		AcceptAllPermissions: d.Get("grant_all_permissions").(bool),
+		Disabled:             d.Get("disabled").(bool),
+		Args:                 getDockerPluginArgs(d.Get("args")),
+	})
```

Comment:
You should implement AcceptPermissionsFunc that matches against permissions listed in a resource attribute that ConfictsWith "grant_all_permissions"


https://pkg.go.dev/github.com/docker/docker/api/types#PluginInstallOptions

Reply:
I don't know why this comment is being hidden but it is vital to the initial implementation. For production use grant_all_permissions must be false and the permissions need to be explicitly provided.

Reply:
@suzuki-shunsuke can you take a look, please

Reply:
I have added an attribute `grant_permissions`.

58fccc62e14bd33702e696971771eeb5f518bcd6

TODO

* test
* document